### PR TITLE
Upgrade the demo app

### PR DESCRIPTION
**Pull request by rohan**
This reopens #29 for review.

### CODE REVIEW

**Review by francoisneron — APPROVED** [review]
This is actually a demo with your own logic and it's pretty straight forward.

**Comment by francoisneron** [comment]
@rohan Let's make sure we get those closed tomorrow, so we can update cocoapods. We have some merchants waiting on this.

**Comment by farhadn** [comment]
Isn't the demo project pulled via cocoapods from this repo https://github.com/Affirm/affirm-ios-sdk-demo ? If so, we shouldn't be making changes to the demo in the current repo.

**Comment by francoisneron** [comment]
@rohan, this is related to some change that were no longer valid on the demo I think.
https://github.com/Affirm/affirm-ios-sdk/commit/4cc69897555db16d03140d931a6b8f24b15d5891

File: AffirmSDK/AffirmAsLowAsButton.h [hunk 1]
```diff
@@ -14,6 +14,7 @@ NS_ASSUME_NONNULL_BEGIN
 @interface AffirmAsLowAsButton : UIButton
 
 @property (nonatomic, weak) id presentingViewController;
+@property (nonatomic, strong) NSString *promoID;
```
Comment (farhadn) [inline AffirmSDK/AffirmAsLowAsButton.h]: This is a change to the SDK and not the demo app. Can we keep this PR only for the demo app and make this change in another PR if required?

Comment (rohan) [reply AffirmSDK/AffirmAsLowAsButton.h]: Done—see #34

File: AffirmSDKDemo/AffirmSDKDemo/ViewController.m [hunk 2]
```diff
@@ -21,31 +28,37 @@ @implementation ViewController
...
 }
 
-- (void)textFieldDidBeginEditing:(UITextField *)textField {
-    self.textField.text = @"";
+- (BOOL)textFieldShouldReturn:(UITextField *)textField {
+    return [textField resignFirstResponder];
 }
 
 - (void)textFieldDidEndEditing:(UITextField *)textField {
+    switch (textField.tag) {
```
Comment (farhadn) [inline AffirmSDKDemo/AffirmSDKDemo/ViewController.m:52]: FYI you don't need to use a tag like this. Can just check `if (textfield == self.amountField)` since the textfield that ended editing is a given parameter.

**Review by farhadn — APPROVED** [review]
